src/access.c

```diff
@@ -361,7 +353,7 @@ int dcc_check_address(const struct sockaddr *client,
         if (value->family == AF_INET) {
             if (IN6_IS_ADDR_V4MAPPED(a6) || IN6_IS_ADDR_V4COMPAT(a6)) {
                 struct in_addr a4;
-                a4.s_addr = inet_from_inet6(a6);
+                inet_from_inet6(&a4, a6);
```

Comment:
This seems to leave the rest of the in_addr uninitialized, which might not matter in this case but seems like a bad idea. Could you zero it or set the fields you know?

Otherwise this looks good, thanks!

Reply:
Actually I'll do that and then merge. Thanks for the patch!

Reply:
> This seems to leave the rest of the in_addr uninitialized, which might not matter in this case but seems like a bad idea. Could you zero it or set the fields you know?

Done